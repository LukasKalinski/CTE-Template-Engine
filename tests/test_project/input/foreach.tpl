{*
  Things to think about when a foreach test fails:
  - Block process ID might differ for some reason.
*}

{* Test foreach with all attributes applied. Also test access to the foreach
   system variable and the variable alias $myItem. *}
{foreach source=$persons id='f1' key='myKey' item='myItem' enable='iteration'}
  {$cte.foreach.f1.iteration}
  {$myKey}
  {$myItem}
{foreachelse}
{/foreach}

{* Test foreach with one set of necessary attributes, expecting 
   shorter output since the foreach system variable doesn't need
   to have additional data set. *}
{foreach source=$persons item='myItem'}
  {$myItem}
{/foreach}

{* Test foreach with another set of necessary attributes (differing from the 
   above). We're expecting it to work and the foreach system variable to be
   set. *}
{foreach id='myForeach' source=$persons}
  {$cte.foreach.myForeach.value}
{/foreach}