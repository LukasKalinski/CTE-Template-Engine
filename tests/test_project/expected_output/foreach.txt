<?php if(isset($this->v_initial['persons'])&&(is_array($this->v_initial['persons'])||is_object($this->v_initial['persons']))&&count($this->v_initial['persons'])){$this->v_foreach[1]['f1']['iteration']=1;foreach($this->v_initial['persons'] as $this->v_foreach[1]['f1'][0]['myKey']=>$this->v_foreach[1]['f1'][0]['myItem']){?>
  <?php echo $this->v_foreach[1]['f1']['iteration'];?>
  <?php echo $this->v_foreach[1]['f1'][0]['myKey'];?>
  <?php echo $this->v_foreach[1]['f1'][0]['myItem'];?>
<?php }}else{?>
<?php }?>

<?php if(isset($this->v_initial['persons'])&&(is_array($this->v_initial['persons'])||is_object($this->v_initial['persons']))&&count($this->v_initial['persons'])){foreach($this->v_initial['persons'] as $this->v_foreach[1][3][0]['myItem']){?>
  <?php echo $this->v_foreach[1][3][0]['myItem'];?>
<?php }}?>

<?php if(isset($this->v_initial['persons'])&&(is_array($this->v_initial['persons'])||is_object($this->v_initial['persons']))&&count($this->v_initial['persons'])){foreach($this->v_initial['persons'] as $this->v_foreach[1]['myForeach']['value']){?>
  <?php echo $this->v_foreach[1]['myForeach']['value'];?>
<?php }}?>